dashboard/dashboard-front/src/components/jobs/deployment/DeployForm.vue

```diff
@@ -0,0 +1,96 @@
+<script setup lang="ts">
+import { ref, type Ref } from 'vue'
+import { useRouter } from 'vue-router'
+import yaml from 'js-yaml'
+import { apiClient } from '@/services/ApiClient'
+import { toastService } from '@/services/ToastService'
+import { progressService } from "@/services/ProgressService"
+
+const yamlManifestRef: Ref<string> = ref('')
+const gitUsername = ref('')
+const gitPassword = ref('')
+const forceEnabled: Ref<boolean> = ref(false)
+const loading = ref(false)
+
+const router = useRouter()
+
+interface CredentialsModel {
+    username: string
+    password: string
+}
+
+function deployJob() {
+    let manifestDict: Record<string, any>
+    try {
+        manifestDict = yaml.load(yamlManifestRef.value) as Record<string, any>
+        if (typeof manifestDict !== 'object') {
+            throw new TypeError('Expected mapping object')
+        }
+    } catch(err: any) {
+        toastService.showErrorDetails('Invalid Manifest YAML', err)
+        return
+    }
+
+    let gitCredentials: CredentialsModel | null = null
+    if (gitUsername.value !== '' || gitPassword.value !== '') {
```

Comment:
`!gitUsername.value || !gitPassword.value` should also work.

Does this check make sense? I might be wrong but you use both the values if the conditional is true and yet you use `||`. Wouldn't `&&` make more sense?